### PR TITLE
Add Flip method to flip boolean value

### PR DESCRIPTION
**Pull request by barryz**
#2 implemented

### CODE REVIEW

File: bool.go [hunk 1]
```diff
@@ -49,6 +49,11 @@ func (ab *AtomicBool) SetTo(yes bool) {
 	}
 }
 
+// Flip toggles the Boolean (replaces with its opposite value).
+func (ab *AtomicBool) Flip() {
+	atomic.StoreInt32((*int32)(ab), atomic.LoadInt32((*int32)(ab))^1)
```
Comment (tevino) [inline bool.go]: This is not atomic.

Comment (barryz) [reply bool.go]: I haven't found a solution yet. 

This PR should be blocked until I find a solution, or you can close it.

**Comment by tevino** [comment]
@barryz 

According to the spec about [Integer overflow](https://golang.org/ref/spec#Integer_overflow).

You could implement `Toggle` by adding 1 to the integer, the Boolean value could then be determined by the last bit(1 for True, 0 for False) of it, a simple `& 1` will work.

**Comment by LeoHChen** [comment]
@tevino , can you merge this PR?

**Comment by tevino** [comment]
> @tevino , can you merge this PR?

I'd love to, but the implementation is not complete, I'll finish it later on.

File: bool.go [hunk 2]
```diff
@@ -49,6 +49,11 @@ func (ab *AtomicBool) SetTo(yes bool) {
 	}
 }
 
+// Toggle negates boolean atomically and returns a new AtomicBool object which holds previous boolean value.
+func (ab *AtomicBool) Toggle() *AtomicBool {
+	return NewBool(atomic.AddInt32((*int32)(ab), 1)&1 == 0)
```
Comment (tevino) [inline bool.go]: The implementation is now completed.

However, I'm considering to change the returning type of `Toggle()` to simple `bool` to keep the consistency with `SetToIf()`.

@barryz What do you think? Can you elaborate on your considerations when you make it an AtomicBool instead?

Opinions are welcome @LeoHChen

**Comment by tevino** [comment]
Since no reply from the original author(@barryz), I'm merging this with a simple bool as the returning type of `Toggle` to make it consistent with the existing APIs.

@barryz and anyone who see this, please feel free to open an issue to talk about [it](https://github.com/tevino/abool/pull/3#discussion_r447373243).

**Comment by tevino** [comment]
@barryz Thanks for your contribution.

**Comment by tevino** [comment]
> @tevino , can you merge this PR?

@LeoHChen You may now use `Toggle` from the latest master.

**Comment by tevino** [comment]
This feature is reverted temporarily due to a regression introduced to `SetToIf`, @LeoHChen if you are using it, please check it out here: https://github.com/tevino/abool/issues/7#issuecomment-913086478

**Pull request merged**